incubator/binding-grpc-kafka/src/main/java/io/aklivity/zilla/runtime/binding/grpc/kafka/internal/config/GrpcKafkaConditionConfigAdapter.java

```diff
@@ -105,22 +105,31 @@ public ConditionConfig adaptFromJson(
             metadata.forEach((k, v) ->
             {
                 final String8FW key = new String8FW(k);
-                final JsonObject value = getJsonObject(v);
-
                 String textValue = null;
                 String base64Value = null;
+                JsonValue.ValueType valueType = v.getValueType();
 
-                if (value != null && value.containsKey(BASE64_NAME))
-                {
-                    base64Value = value.getString(BASE64_NAME);
-                }
-                else
+                switch (valueType)
                 {
-                    textValue = JsonString.class.cast(v).getString();
+                case OBJECT:
+                    if (v.asJsonObject().containsKey(BASE64_NAME))
+                    {
+                        base64Value = v.asJsonObject().getString(BASE64_NAME);
+                    }
+                    else
+                    {
+                        throw new IllegalArgumentException("Unexpected metadata type: " + v.asJsonObject());
+                    }
```

Comment:
I think this validation code can never execute due to schema checks, right?

Reply:
Yeah, schema validation